Sources/Orbit/Support/Components/BarButton.swift

```diff
@@ -3,36 +3,66 @@ import SwiftUI
 /// Button for NavigationBar.
 public struct BarButton: View {
 
-    private let symbol: Icon.Symbol
+    private let content: Icon.Content
     private let action: () -> Void
 
     public var body: some View {
-        SwiftUI.Button(
-            action: {
-                HapticsProvider.sendHapticFeedback(.light(0.5))
-                action()
-            },
-            label: {
-                Icon(symbol, size: .large)
-            }
-        )
-        .buttonStyle(NavigateButton.OrbitStyle())
+        SwiftUI.Button {
+            HapticsProvider.sendHapticFeedback(.light(0.5))
+            action()
+        } label: {
+            Icon(content: content)
+        }
+        .buttonStyle(.barButton(color: content.color))
     }
 
-    public init(_ symbol: Icon.Symbol, action: @escaping () -> Void = {}) {
-        self.symbol = symbol
+    public init(_ content: Icon.Content, action: @escaping () -> Void = {}) {
+        self.content = content
         self.action = action
     }
 }
 
 // MARK: - Types
-private extension NavigateButton {
+private extension BarButton {
+
+    struct ButtonStyle: SwiftUI.ButtonStyle {
 
-    struct OrbitStyle: ButtonStyle {
+        var color: Color?
 
         func makeBody(configuration: Configuration) -> some View {
             configuration.label
-                .foregroundColor(configuration.isPressed ? .inkLight : .inkDark)
+                .foregroundColor(foregroundColor(configuration: configuration))
+        }
+
+        func foregroundColor(configuration: Configuration) -> Color? {
+            let foregroundColor: Color
+
+            if let color = color {
+                foregroundColor = color
+            } else {
+                foregroundColor = .inkDark
+            }
+
+            return configuration.isPressed ? foregroundColor.opacity(0.6) : foregroundColor
+        }
+    }
+}
+
+extension SwiftUI.ButtonStyle where Self == BarButton.ButtonStyle {
```

Comment:
Shouldn't this be public?

Reply:
I thought about keeping it internal, so that the primary way to use this component would be through the `BarButton` view itself, in order not to pollute the SwiftUI.ButtonStyle namespace to call site.